### ISSUE TITLE
Invalid host name

### ISSUE THREAD

**Issue by czebiniak** [headwaymaps/headway#186]

**Comment by michaelkirk** [comment]
> I get the following error in the docker-compose log:

```
headwaymaps-valhalla_init-1 | http://: Invalid host name.
headwaymaps-valhalla_init-1 | xz: (stdin): File format not recognized
```

The container named `headwaymaps-valhalla_init` corresponds to services/valhalla/init.sh  

Looking at that file, it seems likely that you're trying to download $VALHALLA_ARTIFACT_URL. But that shouldn't happen if you've built all the artifacts as the instructions describe.

>  I do have Earthly installed and running, but when I try to run the docker-compose, 

Did you run the build? Something like:

```
earthly -P +build --area=Amsterdam
```

Did it succeed?

**Comment by czebiniak** [comment]
Yes I have tried running that, this is what I always get:

+pelias-import *failed* | fff0edd27e5e: Pull complete
      +pelias-import *failed* | e13dc882f523: Pull complete
      +pelias-import *failed* | 91d2fd5e3a1d: Pull complete
      +pelias-import *failed* | 5c9407b5b09d: Pull complete
      +pelias-import *failed* | Digest: sha256:2dd1b31bc70d91aed2898c1e5fbda4f7de6ef43a9b6d2cad53b1e4c82baa409f
      +pelias-import *failed* | Status: Downloaded newer image for 172.30.0.1:8371/sess-cnvdjj1zikynj4z10b860nnuh:img-0
      +pelias-import *failed* | 172.30.0.1:8371/sess-cnvdjj1zikynj4z10b860nnuh:img-0
      +pelias-import *failed* | Untagged: 172.30.0.1:8371/sess-cnvdjj1zikynj4z10b860nnuh:img-0
      +pelias-import *failed* | Untagged: 172.30.0.1:8371/sess-cnvdjj1zikynj4z10b860nnuh@sha256:2dd1b31bc70d91aed2898c1e5fbda4f7de6ef43a9b6d2cad53b1e4c82baa409f
      +pelias-import *failed* | ...done
      +pelias-import *failed* | Creating network "config_default" with the default driver
      +pelias-import *failed* | Creating volume "config_pelias_data" with default driver
      +pelias-import *failed* | Pulling pelias_elasticsearch (pelias/elasticsearch:7.16.1)...
      +pelias-import *failed* | 7.16.1: Pulling from pelias/elasticsearch
      +pelias-import *failed* | Digest: sha256:570251377c1e17f2b87a0ab27f0e0c7494dd2d4a0815da1b065b9818ea060d2d
      +pelias-import *failed* | Status: Downloaded newer image for pelias/elasticsearch:7.16.1
      +pelias-import *failed* | Creating pelias_elasticsearch ...
      +pelias-import *failed* | Creating pelias_elasticsearch ... done
      +pelias-import *failed* | Creating pelias_schema        ...
      +pelias-import *failed* | Creating pelias_schema        ... done
      +pelias-import *failed* | Creating config_pelias_schema_run ...
      +pelias-import *failed* | Creating config_pelias_schema_run ... done
      +pelias-import *failed* | waiting for elasticsearch service to come up
      +pelias-import *failed* | ..............................

      +pelias-import *failed* | Elasticsearch did not come up, check configuration
      +pelias-import *failed* | 1
      +pelias-import *failed* | Stopping pelias_elasticsearch ...
      +pelias-import *failed* | Stopping pelias_elasticsearch ... done
      +pelias-import *failed* | Removing config_pelias_schema_run_67c3d2d2a8eb ...
      +pelias-import *failed* | Removing pelias_schema                         ...
      +pelias-import *failed* | Removing pelias_elasticsearch                  ...
      +pelias-import *failed* | Removing pelias_elasticsearch                  ... done
      +pelias-import *failed* | Removing pelias_schema                         ... done
      +pelias-import *failed* | Removing config_pelias_schema_run_67c3d2d2a8eb ... done
      +pelias-import *failed* | Removing network config_default
      +pelias-import *failed* | ERROR Earthfile line 262:8
      +pelias-import *failed* |       The command
      +pelias-import *failed* |           WITH DOCKER RUN --privileged docker-compose run -T 'pelias_schema' bash -c "/tools/wait.sh && ./bin/create_index"
      +pelias-import *failed* |       did not complete successfully. Exit code 1
Share your logs with an Earthly account (experimental)! Register for one at https://ci.earthly.dev.
Error: build target: build main: failed to solve: Earthfile line 26:4 apply BUILD +save: earthfile2llb for +save: Earthfile line 41:4 apply BUILD +save-elasticsearch: earthfile2llb for +save-elasticsearch: Earthfile line 96:4 copy artifact: apply build +pelias-import: earthfile2llb for +pelias-import: Earthfile line 262:8 with docker run: unlazy force execution: process "/bin/sh -c EARTHLY_DOCKERD_DATA_ROOT=\"/var/earthly/dind/1e6fcfb633d02e5dda9323eb729bc6993d1f6e13780d30c6e4f451b8c279989f\" EARTHLY_DOCKER_LOAD_FILES=\"\" EARTHLY_IMAGES_WITH_DIGESTS=\"\" EARTHLY_START_COMPOSE=\"true\" EARTHLY_COMPOSE_FILES=\"compose.yaml\" EARTHLY_COMPOSE_SERVICES=\"pelias_schema\" DATA_DIR=/data DIND_COMMIT=42b1175eda071c0e9121e1d64345928384a93df1 DOCKER_TLS_CERTDIR=/certs DOCKER_VERSION=20.10.14 EARTHLY_LOCALLY=false PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin area=NY_PA countries=US /var/earthly/dockerd-wrapper.sh execute /usr/bin/earth_debugger /bin/sh -c 'docker-compose run -T '\"'\"'pelias_schema'\"'\"' bash -c \"/tools/wait.sh && ./bin/create_index\"'" did not complete successfully: exit code: 1
in              +pelias-import --HEADWAY_AREA=NY_PA --HEADWAY_BBOX=41.31 -77.82 42.96 -74.34 --HEADWAY_ENABLE_TRANSIT_ROUTING=0 --HEADWAY_HTTP_PORT=8080 --HEADWAY_PUBLIC_URL=http://localhost:8080 --HOSTNAME=localhost --area=NY_PA --countries=US --transit_feeds=
in              +save-elasticsearch --HEADWAY_AREA=NY_PA --HEADWAY_BBOX=41.31 -77.82 42.96 -74.34 --HEADWAY_ENABLE_TRANSIT_ROUTING=0 --HEADWAY_HTTP_PORT=8080 --HEADWAY_PUBLIC_URL=http://localhost:8080 --HOSTNAME=localhost --area=NY_PA --countries=US --transit_feeds=
in              +save --HEADWAY_AREA=NY_PA --HEADWAY_BBOX=41.31 -77.82 42.96 -74.34 --HEADWAY_ENABLE_TRANSIT_ROUTING=0 --HEADWAY_HTTP_PORT=8080 --HEADWAY_PUBLIC_URL=http://localhost:8080 --HOSTNAME=localhost --area=NY_PA --countries=US --transit_feeds=
in              +build --HEADWAY_AREA=NY_PA --HEADWAY_BBOX=41.31 -77.82 42.96 -74.34 --HEADWAY_ENABLE_TRANSIT_ROUTING=0 --HEADWAY_HTTP_PORT=8080 --HEADWAY_PUBLIC_URL=http://localhost:8080 --HOSTNAME=localhost --area=NY_PA --countries=US

**Comment by czebiniak** [comment]
Also, I have a custom OSM file which is called NY_PA.osm.pbf which I got from BBBIKE, which is in the top level directory as specified

**Comment by ellenhp** [comment]
NY+PA is a big extract, so I'd suspect memory pressure as a potential reason elasticsearch might not come up during import. Try setting your buildkit parallelism to 1 according to FULL_PLANET.md. Just a hunch :)

**Comment by ellenhp** [comment]
Looks like I may not have documented buildkit parallelism, but you can follow [this documentation](https://docs.earthly.dev/docs/earthly-config) to set up an earthly config file. Glancing through the docs it looks like running `earthly config global.buildkit_max_parallelism 1` might also work.

**Comment by czebiniak** [comment]
Okay, I'll try tweaking the settings as suggested, know that I'm running this on a machine with a 5950x, and 64GB of RAM, And I have 4 TB of space free, building on a 1 TB SSD, so resources shouldn't be an issue, but timeouts due to sheer size of data is definitely plausible... If it fails again I'll try with a smaller build, probably a prebuild and see if it works or gets farther. Thanks for the suggestions

**Comment by ellenhp** [comment]
It's pretty easy to OOM while working with large extracts or planet builds if you allow earthly to run many tasks in parallel. I've done it even with 128gb of RAM. I don't think it's a timeout issue but that's certainly possible too. Super curious to see if the buildkit parallelism tweak fixes it. Failing that, another good data point would be building a small area like Amsterdam.

**Comment by czebiniak** [comment]
I tried the parallelism tweak, and it did start elasticsearch, seemed to get further, but then failed out later on with some generic compilation error, seems like it may have to do with permissions when I look more into the issue on. I watched my RAM usage and it only went up a percent or 2 while building, and CPU barely even moved as well... Where does it process the data? I was assuming it was doing things in the current working directory, but I'm not seeing it create any files or anything in the directory I call the build command in... Seems like it's not actually processing any data for some reason. I also tried using supported build NewYork and same thing, says elasticsearch couldn't start.
